functions.php

```diff
@@ -1,35 +1,27 @@
 <?php
 
-function ask() {
-  header('WWW-Authenticate: Basic realm="Mozilla Corporation - LDAP Login"');
-}
-
 function wail_and_bail() {
   header('HTTP/1.0 401 Unauthorized');
-  ask();
   print "<h1>401 Unauthorized</h1>";
   die;
 }
 
+function server_error_and_bail() {
+  header('HTTP/1.0 500 Server Error');
```

Comment:
This is intentional, so that we can identify based on a given 500 code *precisely* what generated that error, and is permitted by RFC:

```
The Reason-Phrase is intended to give a short textual description of the Status-Code. The Status-Code is intended for use by automata and the Reason-Phrase is intended for the human user.
```